axelor-account/src/main/java/com/axelor/apps/account/service/payment/invoice/payment/InvoicePaymentValidateServiceImpl.java

```diff
@@ -335,45 +337,37 @@ public Move fillMoveWithFinancialDiscount(
     Invoice invoice = invoicePayment.getInvoice();
     Company company = invoice.getCompany();
     AccountConfig accountConfig = accountConfigService.getAccountConfig(company);
-
+    Tax financialDiscountTax = null;
+    Account financialDiscountAccount = null;
     if (invoice.getOperationTypeSelect() == InvoiceRepository.OPERATION_TYPE_SUPPLIER_PURCHASE) {
-
-      Account purchAccount = new Account();
-
-      for (AccountManagement accountManagement :
-          accountConfig.getPurchFinancialDiscountTax().getAccountManagementList()) {
-        if (accountManagement.getCompany().equals(company)) {
-          purchAccount = accountManagement.getFinancialDiscountAccount();
-        }
-      }
-
-      move =
-          fillMoveLinesWithFinancialDiscount(
-              invoicePayment,
-              move,
-              customerAccount,
-              purchAccount,
-              accountConfigService.getAccountConfig(company).getPurchFinancialDiscountAccount());
-
+      financialDiscountTax = accountConfig.getPurchFinancialDiscountTax();
+      financialDiscountAccount = accountConfig.getPurchFinancialDiscountAccount();
     } else if (invoice.getOperationTypeSelect() == InvoiceRepository.OPERATION_TYPE_CLIENT_SALE) {
+      financialDiscountTax = accountConfig.getSaleFinancialDiscountTax();
+      financialDiscountAccount = accountConfig.getSaleFinancialDiscountAccount();
+    }
+
+    if (financialDiscountTax == null
+        || financialDiscountAccount == null
+        || CollectionUtils.isEmpty(financialDiscountTax.getAccountManagementList())) {
+      return move;
+    }
 
-      Account saleAccount = new Account();
-      for (AccountManagement accountManagement :
-          accountConfig.getSaleFinancialDiscountTax().getAccountManagementList()) {
-        if (accountManagement.getCompany().equals(company)) {
-          saleAccount = accountManagement.getFinancialDiscountAccount();
-        }
+    Account account = null;
+    for (AccountManagement accountManagement : financialDiscountTax.getAccountManagementList()) {
+      if (company.equals(accountManagement.getCompany())) {
+        account = accountManagement.getFinancialDiscountAccount();
       }
+    }
 
-      move =
-          fillMoveLinesWithFinancialDiscount(
-              invoicePayment,
-              move,
-              customerAccount,
-              saleAccount,
-              accountConfigService.getAccountConfig(company).getSaleFinancialDiscountAccount());
+    if (account == null) {
```

Comment:
Please, add a method into AccountManagmentService to get the correct value. Also, if null throw an exception instead of return a partial move